dask/backends.py

```diff
@@ -119,7 +120,15 @@ def decorator(
 
             @wraps(fn)
             def wrapper(*args, **kwargs):
-                return getattr(self, dispatch_name)(*args, **kwargs)
+                func = getattr(self, dispatch_name)
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    raise RuntimeError(
+                        f"An error occurred while calling the {funcname(func)} "
+                        f"method registered to the {backend} backend.\n"
```

Comment:
```suggestion
                        f"method registered to the {self.backend} backend.\n"
```
I think this will do the trick